src/basic/fileio.c

```diff
@@ -421,6 +430,7 @@ int read_virtual_file(const char *filename, size_t max_size, char **ret_contents
                 buf = malloc(size + 1);
```

Comment:
Maybe we can use `realloc()` here and drop `mfree()` below??

Reply:
realloc() is a technique to hopefully avoid copying memory around when we need to retain the data but need a larger buffer. If we don't need to retain the data, then free() + malloc() should almost always be better, since it gives the allocator the freedom to place the allocation at a totally different place (which is good, given that many allocators place stuff in different pools/places based on allocation size) without this resulting in needless memory copying. here we don't want the data to be kept around, hence we should use free() + malloc().

Reply:
Ah, yeah, indeed. We do not need to copy the previous incomplete data. Thank you for the explanation.